src/EditorFeatures/CSharpTest/ImplementInterface/ImplementInterfaceTests.cs

```diff
@@ -8984,5 +8984,122 @@ public int Foo
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementInterface)]
+        public async Task TestStaticAbstractInterfaceOperator_OnlyExplicitlyImplementable()
+        {
+            await TestInRegularAndScriptAsync(@"
+interface ITest
+{
+    static abstract int operator -(ITest x);
+}
+
+class C : [|ITest|]
+{
+}
+",
+@"
+interface ITest
+{
+    static abstract int operator -(ITest x);
+}
+
+class C : ITest
+{
+    static int ITest.operator -(ITest x)
+    {
+        throw new System.NotImplementedException();
+    }
+}
+", parseOptions: CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.Preview), index: 0, title: FeaturesResources.Implement_all_members_explicitly);
+        }
+
+        [WorkItem(53927, "https://github.com/dotnet/roslyn/issues/53927")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementInterface)]
+        public async Task TestStaticAbstractInterfaceOperator_ImplementImplicitly()
+        {
+            await TestInRegularAndScriptAsync(@"
+interface ITest<T> where T : ITest<T>
+{
+    static abstract int operator -(T x);
+}
+
+class C : [|ITest<C>|]
+{
+}
+",
+@"
+interface ITest<T> where T : ITest<T>
+{
+    static abstract int operator -(T x);
+}
+
+class C : ITest<C>
+{
+    public static int operator -(C x)
+    {
+        throw new System.NotImplementedException();
+    }
+}
+", parseOptions: CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.Preview), index: 0, title: FeaturesResources.Implement_interface);
+        }
+
+        [WorkItem(53927, "https://github.com/dotnet/roslyn/issues/53927")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementInterface)]
+        public async Task TestStaticAbstractInterfaceOperator_ImplementExplicitly()
+        {
+            await TestInRegularAndScriptAsync(@"
+interface ITest<T> where T : ITest<T>
+{
+    static abstract int operator -(T x);
+}
+
+class C : [|ITest<C>|]
+{
+}
+",
+@"
+interface ITest<T> where T : ITest<T>
+{
+    static abstract int operator -(T x);
+}
+
+class C : ITest<C>
+{
+    static int ITest<C>.operator -(C x)
+    {
+        throw new System.NotImplementedException();
+    }
+}
+", parseOptions: CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.Preview), index: 1, title: FeaturesResources.Implement_all_members_explicitly);
+        }
+
+        [WorkItem(53927, "https://github.com/dotnet/roslyn/issues/53927")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementInterface)]
+        public async Task TestStaticAbstractInterfaceOperator_ImplementAbstractly()
+        {
+            await TestInRegularAndScriptAsync(@"
+interface ITest<T> where T : ITest<T>
+{
+    static abstract int operator -(T x);
+}
+
+abstract class C : [|ITest<C>|]
+{
+}
+",
+@"
+interface ITest<T> where T : ITest<T>
+{
+    static abstract int operator -(T x);
+}
+
+abstract class C : ITest<C>
+{
+    public abstract static int operator -(C x);
```

Comment:
i'm curious.  is this legal?

Reply:
Abstract statics are allowed only in interfaces

Reply:
Thanks for noting @CyrusNajmabadi @AlekseyTs 
I'm curious whether the "Implement interface abstractly" should do nothing for the operator, or generate it without the "abstract" keyword?

Reply:
implement abstractly needs to take the code from uncompilable state to compilable, attempting for each member it *needs to* generate to make that abstract.  If the member doesn't need to be generated, we should not generate it.

Reply:
@CyrusNajmabadi The member needs to be generated to be in a compilable state, but it cannot be generated abstractly.

Reply:
I do not believe there is away to implement a static member abstractly. So, if there are static abstract members in the interface, the option probably shouldn't be offered.